src/features/events/components/EventActionButton/index.tsx

```diff
@@ -1,78 +1,29 @@
-import { useRouter } from 'next/router'
 import type { FC } from 'react'
 
-import type { UserType } from '~/features/auth/contexts/userContext'
-
-import { StyledButton } from './styled'
+import { useUserContext } from '~/features/auth/contexts/userContext'
+import type { ArticleType } from '~/features/events/types'
+import { Button } from '~/features/ui/components/Button'
 
 import { useAttendance } from '../../hooks/useAttendance'
+import { isUserAttending } from '../../lib/isUserAttending'
 
 type Props = {
-  isLoggedInUserOwner: boolean
-  isLoggedInUserAttending: boolean
-  eventID: string
-  isPast: boolean
-  user: UserType | null
-}
-
-type ButtonProps = {
-  buttonText: 'EDIT' | 'JOIN' | 'LEAVE'
-  accent: 'primary' | 'destructive' | 'silent'
-}
-
-const getButtonInfo = (
-  isLoggedInUserOwner: boolean,
-  isLoggedInUserAttending: boolean
-): ButtonProps => {
-  if (isLoggedInUserOwner) {
-    return { buttonText: 'EDIT', accent: 'silent' }
-  }
-  if (isLoggedInUserAttending) {
-    return { buttonText: 'LEAVE', accent: 'destructive' }
-  }
-
-  return { buttonText: 'JOIN', accent: 'primary' }
+  event: ArticleType
 }
```

Comment:
I get that the idea here was to create some sort of a map. But, by having two different concerns – ownership and attendance – this map is simply trying to encompass two different "languages".